dev/com.ibm.ws.logging/src/com/ibm/ws/logging/source/LogSource.java

```diff
@@ -109,28 +122,66 @@ public void publish(RoutedMessage routedMessage) {
         }
     }
 
-    public MessageLogData parse(RoutedMessage routedMessage, LogRecord logRecord) {
-        String message = routedMessage.getFormattedVerboseMsg();
-        if (message == null)
-            message = logRecord.getMessage();
-        long date = logRecord.getMillis();
-        String messageId = null;
-        if (message != null)
-            messageId = parseMessageId(message);
-        int threadId = (int) Thread.currentThread().getId();//logRecord.getThreadID();
-        String loggerName = logRecord.getLoggerName();
-        String logLevel = LogFormatUtils.mapLevelToType(logRecord);
-        String logLevelRaw = LogFormatUtils.mapLevelToRawType(logRecord);
-        String methodName = logRecord.getSourceMethodName();
-        String className = logRecord.getSourceClassName();
-        Map<String, String> extensions = null;
-        if (logRecord instanceof WsLogRecord)
-            extensions = ((WsLogRecord) logRecord).getExtensions();
-        String sequence = sequenceNumber.next(date);
+    private String extractMessage(RoutedMessage routedMessage, LogRecord logRecord) {
+        String messageVal = routedMessage.getFormattedVerboseMsg();
+        if (messageVal == null) {
+            messageVal = logRecord.getMessage();
+        }
+        return messageVal;
+    }
+
+    public GenericData parse(RoutedMessage routedMessage) {
+
+        GenericData genData = new GenericData();
+        LogRecord logRecord = routedMessage.getLogRecord();
+        String messageVal = extractMessage(routedMessage, logRecord);
+
+        long dateVal = logRecord.getMillis();
+        // Must pass datetime as string, as its value type must be set as string in its kvp
+        genData.addPair("ibm_datetime", Long.toString(dateVal));
```

Comment:
just add ibm_datetime as a long dateVal... it's formatter's job to decide if we want it to be a formatted like a string.